### PR TITLE
Add debugging section for Python

### PR DESCRIPTION
**Pull request by JesterOrNot**

### CODE REVIEW

**Comment by JesterOrNot** [comment]
PythonDebug

File: src/docs/python-in-gitpod.md [hunk 1]
```diff
@@ -86,3 +86,12 @@ vscode:
   extensions:
     - littlefoxteam.vscode-python-test-adapter@0.3.16:tZ/6xOSSdKUaq6JCUVkD+A==
 ```
+
+## Debugging
+To start debugging your python application in Gitpod please add a new directory `.theia` and inside add another file  `launch.json` here is a quick clip on how to configure debugging for Python!
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]:
```suggestion
To start debugging your Python application in Gitpod, please add a new directory called `.theia/`, and inside add another file called `launch.json`. Here is a quick clip on how to configure debugging for Python!
```

File: src/docs/python-in-gitpod.md [hunk 2]
```diff
@@ -86,3 +86,12 @@ vscode:
   extensions:
     - littlefoxteam.vscode-python-test-adapter@0.3.16:tZ/6xOSSdKUaq6JCUVkD+A==
 ```
+
+## Debugging
+To start debugging your Python application in Gitpod, please add a new directory called `.theia/`, and inside add another file called `launch.json`. Here is a quick clip on how to configure debugging for Python!
+
+![](images/PythonDebug.gif)
+
+To see a basic repository with debugging please try:
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]: As a user, I don't really like instruction like "just click here: [button]", as I'm not really sure what will happen.

Please at least add an additional link to the repository, e.g.

```suggestion
To see a basic repository with Python debugging enabled, please check out [JesterOrNot/Gitpod-Python-Debug](https://github.com/JesterOrNot/Gitpod-Python-Debug):
```

File: src/docs/python-in-gitpod.md [hunk 3]
```diff
@@ -86,3 +86,12 @@ vscode:
   extensions:
     - littlefoxteam.vscode-python-test-adapter@0.3.16:tZ/6xOSSdKUaq6JCUVkD+A==
 ```
+
+## Debugging
+To start debugging your Python application in Gitpod, please add a new directory called `.theia/`, and inside add another file called `launch.json`. Here is a quick clip on how to configure debugging for Python!
+
+![](images/PythonDebug.gif)
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]: Please add a very brief summary of what you're doing in the GIF, e.g.

> Basically: Open the Debug panel, create a new launch configuration, then select "Python" and "Python File".

File: src/docs/python-in-gitpod.md [hunk 4]
```diff
@@ -86,3 +86,47 @@ vscode:
   extensions:
     - littlefoxteam.vscode-python-test-adapter@0.3.16:tZ/6xOSSdKUaq6JCUVkD+A==
 ```
+
+## Debugging
+To start debugging your Python application in Gitpod, please add a new directory called `.theia/`, and inside add a file called `launch.json`, and add the following to it:
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]: Now I'm wondering, should we really tell users to create this manually? Or rather let them automatically create this via the Debug panel as explained below?

I would personally amend this a little to say it like:

> To start debugging your Python application in Gitpod, you will need a `.theia/launch.json` configuration file that looks a bit like this:
> [...]
> Note: You can automatically generate this configuration file by following the steps below.

File: src/docs/python-in-gitpod.md [hunk 5]
```diff
@@ -86,3 +86,47 @@ vscode:
...
+            "request": "launch",
+            "program": "${file}",
+            "console": "internalConsole"
+        }
+    ]
+}
+```
+Then open the file you want to debug, then in this menu:
+
+![](images/toolbar.png)
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]: I find this screenshot not really useful, and it takes a lot of space. Maybe just phrase it like this:

> Simply open the Python file you want to debug, then open the Debug panel (in the left vertical toolbar, click the icon with the crossed-out-spider), and click the green "Run" button.

File: src/docs/python-in-gitpod.md [hunk 6]
```diff
@@ -86,3 +86,47 @@ vscode:
...
+
+click the picture of the crossed out spider and click the green run button.
+
+Here is a quick clip on how to configure debugging for Python!
+
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]: Unnecessary empty line:

```suggestion
```

File: src/docs/python-in-gitpod.md [hunk 7]
```diff
@@ -86,3 +86,47 @@ vscode:
...
+
+![](images/PythonDebug.gif)
+
+So basically in this video we:
+1. Open the Python file we want to debug
+2. Go to Debug and choose add configuration
+3. In dropdown choose Python
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]:
```suggestion
3. In the dropdown choose "Python"
```

File: src/docs/python-in-gitpod.md [hunk 8]
```diff
@@ -86,3 +86,47 @@ vscode:
...
+4. Choose Python File for the configuration
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]:
```suggestion
4. Choose "Python File" for the debug configuration
```

File: src/docs/python-in-gitpod.md [hunk 9]
```diff
@@ -86,3 +86,47 @@ vscode:
...
+5. Start Debugging
+
+To see a basic repository with Python debugging enabled, please check out [JesterOrNot/Gitpod-Python-Debug](https://github.com/JesterOrNot/Gitpod-Python-Debug):
+
+[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/JesterOrNot/Gitpod-Python-Debug)
+
+<br>
+
+## Resources
+* ***[VSCode documentation for Python debugging](https://code.visualstudio.com/docs/python/debugging)*** All information here should apply to Gitpod
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]:
```suggestion
* ***[VSCode documentation for Python debugging](https://code.visualstudio.com/docs/python/debugging)*** All information here should apply to Gitpod as well
```

File: src/docs/python-in-gitpod.md [hunk 10]
```diff
@@ -86,3 +86,47 @@ vscode:
...
+click the picture of the crossed out spider and click the green run button.
+
+Here is a quick clip on how to configure debugging for Python!
+
+
+![](images/PythonDebug.gif)
+
+So basically in this video we:
+1. Open the Python file we want to debug
+2. Go to Debug and choose add configuration
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]:
```suggestion
2. Go to Debug and choose "Add Configuration..."
```

File: src/docs/python-in-gitpod.md [hunk 11]
```diff
@@ -65,24 +86,45 @@ Here are some other examples of Python GUI applications in Gitpod:
...
+![](images/PythonDebug.gif)
+
+So basically in this video we:
+1. First, open the Python file that we want to debug
+2. Then, go to the debug menu and select "Add Configuration..."
+3. Next, in the dropdown choose "Python"
+4. Next, choose "Python File" as the debug configuration
+5. Finally, Start Debugging
+
+How to configure manually
```
Comment (jankeromnes) [inline src/docs/python-in-gitpod.md]: If I'm being nit-picky, I find this title a little bit ambiguous. Maybe clarify? E.g.

> You can also create the Python debug configuration file manually

**Pull request merged**